DAOS-5508 test: more datamover tests

daltonbohning opened the pull request.

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+import os
+import uuid
+
+    
```
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+        """Set up each test case."""
+        # Start the servers and agents
+        super(CopyProcsTest, self).setUp()
+        
```
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+        """Tear down each test case."""
+        # Stop the servers and agents
+        super(CopyProcsTest, self).tearDown()
+    
```
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+
+        # Create a pool
+        pool.create()
+        
```
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+
+        if path is not None:
+            container.path.update(path)
+        
```
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+
+        # Create the test file
+        self.write_daos(pool1, container1)
+        
```
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
```suggestion

```

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+    """
+
+    def __init__(self, *args, **kwargs):
+       	"""Initialize a CopyBasicsTest object."""
```
(pylint-mixed-indentation) Found indentation with tabs instead of spaces

daosbuild1 replied:
(pylint-mixed-indentation) Found indentation with tabs instead of spaces

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+        super(CopyProcsTest, self).setUp()
+        
+        # Get the parameters
+        self.flags_write = self.params.get("flags_write", "/run/ior/copy_procs/*")
```
(pylint-line-too-long) Line too long (82/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (82/80)

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+        
+        # Run with varying number of processes
+        for num_procs in procs_list:
+            test_desc="copy_procs (processes={})".format(num_procs)
```
(pylint-bad-whitespace) Exactly one space required around assignment

daosbuild1 replied:
(pylint-bad-whitespace) Exactly one space required around assignment

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+        for num_procs in procs_list:
+            test_desc="copy_procs (processes={})".format(num_procs)
+            self.run_dcp(
+                src="/", dst ="/",
```
(pylint-bad-whitespace) No space allowed before keyword argument assignment

daosbuild1 replied:
(pylint-bad-whitespace) No space allowed before keyword argument assignment

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+            dcp.run(self.workdir, processes)
+        except CommandFailure as error:
+            self.log.error("DCP command failed: %s", str(error))
+            self.fail("Test was expected to pass but it failed: {}\n".format(test_desc))
```
(pylint-line-too-long) Line too long (88/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (88/80)

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+import uuid
+
+    
+class CopyProcsTest(IorTestBase):
```
(pylint-too-many-ancestors) Too many ancestors (9/7)

daosbuild1 replied:
(pylint-too-many-ancestors) Too many ancestors (9/7)

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+
+        return pool
+
+    def create_cont(self, pool, path=None):
```
(pylint-arguments-differ) Arguments number differs from overridden 'create_cont' method

daosbuild1 replied:
(pylint-arguments-differ) Arguments number differs from overridden 'create_cont' method

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+        """Uses ior to write the test file to a DAOS container."""
+        self.ior_cmd.flags.update(self.flags_write)
+        self.ior_cmd.set_daos_params(self.server_group, pool, container.uuid)
+        out = self.run_ior(self.get_ior_job_manager_command(), self.processes)
```
(pylint-unused-variable) Unused variable 'out'

daosbuild1 replied:
(pylint-unused-variable) Unused variable 'out'

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+        """Uses ior to read-verify the test file in a DAOS container."""
+        self.ior_cmd.flags.update(self.flags_read)
+        self.ior_cmd.set_daos_params(self.server_group, pool, container.uuid)
+        out = self.run_ior(self.get_ior_job_manager_command(), self.processes)
```
(pylint-unused-variable) Unused variable 'out'

daosbuild1 replied:
(pylint-unused-variable) Unused variable 'out'

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+
+        # Run the dcp command
+        if test_desc is not None:
+            self.log.info("Running dcp: {}".format(test_desc))
```
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments

daosbuild1 replied:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+from test_utils_pool import TestPool
+from test_utils_container import TestContainer
+from data_mover_utils import DataMover
+import os
```
(pylint-unused-import) Unused import os

daosbuild1 replied:
(pylint-unused-import) Unused import os

daosbuild1 commented on src/tests/ftest/datamover/copy_procs.py:
```diff
@@ -0,0 +1,163 @@
+from test_utils_container import TestContainer
+from data_mover_utils import DataMover
+import os
+import uuid
```
(pylint-unused-import) Unused import uuid

daosbuild1 replied:
(pylint-unused-import) Unused import uuid

daosbuild1 commented on src/tests/ftest/datamover/copy_subsets.py:
```diff
@@ -0,0 +1,186 @@
+from data_mover_utils import DataMover
+import os
+
+    
```
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 commented on src/tests/ftest/datamover/copy_subsets.py:
```diff
@@ -0,0 +1,186 @@
+class CopySubsetsTest(IorTestBase, MdtestBase):
+    """
+    Test Class Description:
+        Tests ability to copy container subsets. 
```
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 commented on src/tests/ftest/datamover/copy_subsets.py:
```diff
@@ -0,0 +1,186 @@
+        """Set up each test case."""
+        # Start the servers and agents
+        super(CopySubsetsTest, self).setUp()
+        
```
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 commented on src/tests/ftest/datamover/copy_subsets.py:
```diff
@@ -0,0 +1,186 @@
+        cmd = "rm -r '{}'".format(
+            self.uns_dir)
+        self.execute_cmd(cmd)
+        
```
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 commented on src/tests/ftest/datamover/copy_subsets.py:
```diff
@@ -0,0 +1,186 @@
+        
+        # Stop the servers and agents
+        super(CopySubsetsTest, self).tearDown()
+    
```
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 commented on src/tests/ftest/datamover/copy_subsets.py:
```diff
@@ -0,0 +1,186 @@
+
+        if path is not None:
+            container.path.update(path)
+        
```
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 commented on src/tests/ftest/datamover/copy_subsets.py:
```diff
@@ -0,0 +1,186 @@
+        self.mdtest_cmd.flags.update(self.flags_write)
+        self.mdtest_cmd.test_dir.update(test_dir)
+        self.mdtest_cmd.set_daos_params(self.server_group, pool, container.uuid)
+        out = self.run_mdtest(self.get_mdtest_job_manager_command(self.manager), 
```
(style)  trailing whitespace
(pylint-unused-variable) Unused variable 'out'

daosbuild1 replied:
(style)  trailing whitespace
(pylint-unused-variable) Unused variable 'out'

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3762/1/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

Note: Error annotation limited to the first 31 errors.  Remaining unannotated errors:

[src/tests/ftest/datamover/copy_subsets.py:162](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L162):
(style)  trailing whitespace

[src/tests/ftest/datamover/copy_subsets.py:44](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L44):
(pylint-mixed-indentation) Found indentation with tabs instead of spaces

[src/tests/ftest/datamover/copy_subsets.py:55](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L55):
(pylint-line-too-long) Line too long (87/80)

[src/tests/ftest/datamover/copy_subsets.py:56](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L56):
(pylint-line-too-long) Line too long (85/80)

[src/tests/ftest/datamover/copy_subsets.py:57](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L57):
(pylint-line-too-long) Line too long (82/80)

[src/tests/ftest/datamover/copy_subsets.py:181](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L181):
(pylint-bad-indentation) Bad indentation. Found 13 spaces, expected 12
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments

[src/tests/ftest/datamover/copy_subsets.py:186](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L186):
(pylint-line-too-long) Line too long (88/80)

[src/tests/ftest/datamover/copy_subsets.py:34](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L34):
(pylint-too-many-ancestors) Too many ancestors (10/7)

[src/tests/ftest/datamover/copy_subsets.py:74](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L74):
(pylint-arguments-differ) Arguments number differs from overridden 'create_cont' method

[src/tests/ftest/datamover/copy_subsets.py:160](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L160):
(pylint-unused-variable) Unused variable 'out'

[src/tests/ftest/datamover/copy_subsets.py:28](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_subsets.py#L28):
(pylint-unused-import) Unused TestPool imported from test_utils_pool

[src/tests/ftest/datamover/container_copy.py:40](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/container_copy.py#L40):
(pylint-arguments-differ) Arguments number differs from overridden 'create_cont' method

[src/tests/ftest/datamover/copy_basics.py:254](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_basics.py#L254):
(pylint-line-too-long) Line too long (83/80)

[src/tests/ftest/datamover/copy_basics.py:384](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_basics.py#L384):
(pylint-line-too-long) Line too long (88/80)

[src/tests/ftest/datamover/copy_basics.py:417](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_basics.py#L417):
(pylint-too-few-public-methods) Too few public methods (0/2)

[src/tests/ftest/datamover/copy_basics.py:452](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_basics.py#L452):
(pylint-too-few-public-methods) Too few public methods (0/2)

[src/tests/ftest/datamover/copy_negative.py:372](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L372):
(lint) how to limit the space?

[src/tests/ftest/datamover/copy_negative.py:442](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L442):
(pylint-fixme) TODO re-evaluate how to do this.

[src/tests/ftest/datamover/copy_negative.py:54](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L54):
(pylint-mixed-indentation) Found indentation with tabs instead of spaces

[src/tests/ftest/datamover/copy_negative.py:67](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L67):
(pylint-line-too-long) Line too long (85/80)

[src/tests/ftest/datamover/copy_negative.py:69](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L69):
(pylint-line-too-long) Line too long (95/80)

[src/tests/ftest/datamover/copy_negative.py:70](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L70):
(pylint-line-too-long) Line too long (81/80)

[src/tests/ftest/datamover/copy_negative.py:71](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L71):
(pylint-line-too-long) Line too long (83/80)

[src/tests/ftest/datamover/copy_negative.py:74](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L74):
(pylint-line-too-long) Line too long (81/80)

[src/tests/ftest/datamover/copy_negative.py:75](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L75):
(pylint-line-too-long) Line too long (81/80)

[src/tests/ftest/datamover/copy_negative.py:259](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L259):
(pylint-line-too-long) Line too long (81/80)

[src/tests/ftest/datamover/copy_negative.py:463](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L463):
(pylint-line-too-long) Line too long (88/80)

[src/tests/ftest/datamover/copy_negative.py:470](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L470):
(pylint-line-too-long) Line too long (88/80)

[src/tests/ftest/datamover/copy_negative.py:35](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L35):
(pylint-too-many-ancestors) Too many ancestors (9/7)

[src/tests/ftest/datamover/copy_negative.py:45](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L45):
(pylint-pointless-string-statement) String statement has no effect

[src/tests/ftest/datamover/copy_negative.py:115](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L115):
(pylint-arguments-differ) Arguments number differs from overridden 'create_cont' method

[src/tests/ftest/datamover/copy_negative.py:158](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L158):
(pylint-unused-variable) Unused variable 'container2'

[src/tests/ftest/datamover/copy_negative.py:381](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L381):
(pylint-unused-variable) Unused variable 'out'

[src/tests/ftest/datamover/copy_negative.py:392](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L392):
(pylint-unused-variable) Unused variable 'out'

[src/tests/ftest/datamover/copy_negative.py:394](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L394):
(pylint-no-self-use) Method could be a function

[src/tests/ftest/datamover/copy_negative.py:403](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L403):
(pylint-no-self-use) Method could be a function

[src/tests/ftest/datamover/copy_negative.py:411](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L411):
(pylint-too-many-arguments) Too many arguments (17/10)

[src/tests/ftest/datamover/copy_negative.py:458](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L458):
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments

[src/tests/ftest/datamover/copy_negative.py:30](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/datamover/copy_negative.py#L30):
(pylint-unused-import) Unused skipForTicket imported from apricot

[src/tests/ftest/util/data_mover_utils.py:143](https://github.com/daos-stack/daos/blob/fa54a3504423b529ac087286a523e883bc4cb727/src/tests/ftest/util/data_mover_utils.py#L143):
(pylint-arguments-differ) Arguments number differs from overridden 'run' method

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/1/execution/node/60/log

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3762/2/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/2/execution/node/60/log

daosbuild1 commented:
Test stage Functional on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3762/2/display/redirect

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/4/execution/node/214/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/4/execution/node/207/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/4/execution/node/197/log

daltonbohning commented:
Current Status:
dcp seems to working as expected in CI.
Pushed a temporary workaround for ior POSIX permissions, pending results

daltonbohning commented:
Current Status:
dcp seems to working as expected in CI.
There is 1 intermittent test failure, working on it.
The `copy_subsets` tests fail, waiting for a fix to merged into upstream mpifileutils. If this isn't merged soon, I will add skipForTicket and revisit this test case at a later time.

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+        self.run_datamover(
+            test_desc="copy_error_check (dst pool out of space)",
+            expected_rc=1,
+            expected_output=[self.MFU_ERR_DCP_COPY, "errno=28"])        
```
(style)  trailing whitespace

daosbuild1 replied:
```suggestion
            expected_output=[self.MFU_ERR_DCP_COPY, "errno=28"])
```

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_rc=1,
+            expected_output=[self.MFU_ERR_DCP_COPY, "errno=28"])        
+
+        # TODO: how to limit the space for POSIX?
```
(lint) how to limit the space for POSIX?

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+    MFU_ERR_DAOS_INVAL_ARG = "MFU_ERR(-4001)"
+
+    def __init__(self, *args, **kwargs):
+       	"""Initialize a CopyNegativeTest object."""
```
(pylint-mixed-indentation) Found indentation with tabs instead of spaces

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+        super(CopyNegativeTest, self).setUp()
+
+        # Get the parameters
+        self.flags_write = self.params.get("flags_write", "/run/ior/copy_negative/*")
```
(pylint-line-too-long) Line too long (85/80)

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+        # Get the parameters
+        self.flags_write = self.params.get("flags_write", "/run/ior/copy_negative/*")
+        self.block_size = self.params.get("block_size", "/run/ior/*")
+        self.block_size_large = self.params.get("block_size_large", "/run/ior/copy_negative/*")
```
(pylint-line-too-long) Line too long (95/80)

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+        self.flags_write = self.params.get("flags_write", "/run/ior/copy_negative/*")
+        self.block_size = self.params.get("block_size", "/run/ior/*")
+        self.block_size_large = self.params.get("block_size_large", "/run/ior/copy_negative/*")
+        self.test_file = self.params.get("test_file", "/run/ior/copy_negative/*")
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+
+        """ (2) Bad parameter: source is destination.
+            These tests use the same source and destination."""
+        self.set_src_location("DAOS_UUID", "/", pool1, container1);
```
(pylint-unnecessary-semicolon) Unnecessary semicolon

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+        """ (2) Bad parameter: source is destination.
+            These tests use the same source and destination."""
+        self.set_src_location("DAOS_UUID", "/", pool1, container1);
+        self.set_dst_location("DAOS_UUID", "/", pool1, container1);
```
(pylint-unnecessary-semicolon) Unnecessary semicolon

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_rc=1,
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        self.set_src_location("DAOS_UNS", "/", pool1, container1);
```
(pylint-unnecessary-semicolon) Unnecessary semicolon

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        self.set_src_location("DAOS_UNS", "/", pool1, container1);
+        self.set_dst_location("DAOS_UNS", "/", pool1, container1);
```
(pylint-unnecessary-semicolon) Unnecessary semicolon

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_rc=1,
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        self.set_src_location("DAOS_UUID", "/", pool1, container1);
```
(pylint-unnecessary-semicolon) Unnecessary semicolon

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        self.set_src_location("DAOS_UUID", "/", pool1, container1);
+        self.set_dst_location("DAOS_UNS", "/", pool1, container1);
```
(pylint-unnecessary-semicolon) Unnecessary semicolon

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_rc=1,
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        self.set_src_location("DAOS_UNS", "/", pool1, container1);
```
(pylint-unnecessary-semicolon) Unnecessary semicolon

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        self.set_src_location("DAOS_UNS", "/", pool1, container1);
+        self.set_dst_location("DAOS_UUID", "/", pool1, container1);
```
(pylint-unnecessary-semicolon) Unnecessary semicolon

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+from data_mover_test_base import DataMoverTestBase
+from os.path import join, sep
+
+class CopyNegativeTest(DataMoverTestBase):
```
(pylint-too-many-ancestors) Too many ancestors (10/7)

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+    :avocado: recursive
+    """
+
+    """DCP error codes"""
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+
+        """ Bad parameter: required arguments.
+            These tests use the same valid destination parameters,
+            but varying invalid source parameters. """
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+
+        """ Bad parameter: required arguments.
+            These tests use the same valid source parameters,
+            but varying invalid destination parameters."""
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        """ Bad parameter: required arguments.
+            These tests use missing prefix/UNS parameters."""
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        """ (2) Bad parameter: source is destination.
+            These tests use the same source and destination."""
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        """ (3) Bad parameter: daos-prefix is invalid.
+            These tests use invalid prefixes."""
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+            expected_output=self.MFU_ERR_DAOS_INVAL_ARG)
+
+        """ (4) Bad parameter: UUID, UNS, or POSIX path does not exist.
+            These tests use parameters that do not exist. """
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_negative.py:
```diff
@@ -0,0 +1,367 @@
+        :avocado: tags=copy_negative,copy_bad_params
+        """
+        # Start dfuse to hold all pools/containers
+        self.dfuse_hosts = self.agent_managers[0].hosts
```
(pylint-attribute-defined-outside-init) Attribute 'dfuse_hosts' defined outside __init__

daosbuild1 commented on src/tests/ftest/datamover/container_copy.py:
```diff
@@ -0,0 +1,169 @@
+
+        self.container = []
+
+    def create_cont(self):
```
(pylint-arguments-differ) Arguments number differs from overridden 'create_cont' method

daosbuild1 replied:
(pylint-arguments-differ) Arguments number differs from overridden 'create_cont' method

daosbuild1 commented on src/tests/ftest/datamover/copy_basics.py:
```diff
@@ -0,0 +1,356 @@
+            as a directory, and the test file is appended
+            to this path in write/read_location.
+            This allows UNS -> UNS to be performed.
+        """
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_basics.py:
```diff
@@ -0,0 +1,356 @@
+        """ Run and verify each copy.
+            Each src or dst is a list of params:
+                [param_type, path, pool, cont]
+        """
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_basics.py:
```diff
@@ -0,0 +1,356 @@
+        """
+        # Create pools and src container
+        pool1 = self.create_pool()
+        pool2 = self.create_pool()
```
(pylint-unused-variable) Unused variable 'pool2'

daosbuild1 commented on src/tests/ftest/datamover/copy_basics.py:
```diff
@@ -0,0 +1,356 @@
+            For each copy, use a new destination directory.
+            This ensures that the source directory is copied
+            *to* the destination, instead of *into* it.
+        """
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_basics.py:
```diff
@@ -0,0 +1,356 @@
+        """ Run and verify each copy.
+            Each src or dst is a list of params:
+                [param_type, path, pool, cont]
+        """
```
(pylint-pointless-string-statement) String statement has no effect

daosbuild1 commented on src/tests/ftest/datamover/copy_basics.py:
```diff
@@ -0,0 +1,356 @@
+        :avocado: tags=copy_basics,copy_types
+        """
+        # Start dfuse to hold all pools/containers
+        self.dfuse_hosts = self.agent_managers[0].hosts
```
(pylint-attribute-defined-outside-init) Attribute 'dfuse_hosts' defined outside __init__

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3762/13/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

Note: Error annotation limited to the first 31 errors.  Remaining unannotated errors:

[src/tests/ftest/datamover/copy_basics.py:272](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/datamover/copy_basics.py#L272):
(pylint-attribute-defined-outside-init) Attribute 'dfuse_hosts' defined outside __init__

[src/tests/ftest/datamover/copy_procs.py:37](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/datamover/copy_procs.py#L37):
(pylint-mixed-indentation) Found indentation with tabs instead of spaces

[src/tests/ftest/datamover/copy_procs.py:109](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/datamover/copy_procs.py#L109):
(pylint-bad-whitespace) Exactly one space required around assignment

[src/tests/ftest/datamover/copy_procs.py:121](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/datamover/copy_procs.py#L121):
(pylint-line-too-long) Line too long (84/80)
(pylint-bad-whitespace) Exactly one space required around assignment

[src/tests/ftest/datamover/copy_procs.py:27](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/datamover/copy_procs.py#L27):
(pylint-too-many-ancestors) Too many ancestors (10/7)

[src/tests/ftest/util/data_mover_test_base.py:313](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L313):
(pylint-bad-continuation) Wrong continued indentation (add 20 spaces).

[src/tests/ftest/util/data_mover_test_base.py:315](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L315):
(pylint-bad-continuation) Wrong continued indentation (add 26 spaces).

[src/tests/ftest/util/data_mover_test_base.py:318](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L318):
(pylint-bad-continuation) Wrong continued indentation (add 20 spaces).

[src/tests/ftest/util/data_mover_test_base.py:320](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L320):
(pylint-bad-continuation) Wrong continued indentation (add 26 spaces).

[src/tests/ftest/util/data_mover_test_base.py:340](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L340):
(pylint-line-too-long) Line too long (81/80)

[src/tests/ftest/util/data_mover_test_base.py:385](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L385):
(pylint-line-too-long) Line too long (88/80)

[src/tests/ftest/util/data_mover_test_base.py:66](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L66):
(pylint-pointless-string-statement) String statement has no effect

[src/tests/ftest/util/data_mover_test_base.py:179](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L179):
(pylint-no-self-use) Method could be a function

[src/tests/ftest/util/data_mover_test_base.py:184](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L184):
(pylint-unused-argument) Unused argument 'display'

[src/tests/ftest/util/data_mover_test_base.py:189](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L189):
(pylint-unused-argument) Unused argument 'display'

[src/tests/ftest/util/data_mover_test_base.py:235](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L235):
(pylint-pointless-string-statement) String statement has no effect

[src/tests/ftest/util/data_mover_test_base.py:172](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_test_base.py#L172):
(pylint-attribute-defined-outside-init) Attribute '_gen_daos_path_v' defined outside __init__

[src/tests/ftest/util/data_mover_utils.py:143](https://github.com/daos-stack/daos/blob/5ca628ab15b0622f26ba775d436b3382123e1190/src/tests/ftest/util/data_mover_utils.py#L143):
(pylint-arguments-differ) Arguments number differs from overridden 'run' method

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/13/execution/node/57/log

daltonbohning commented:
Current Status:
dcp seems to working as expected in CI.
There is 1 intermittent test failure, working on it.
The copy_subsets tests fail, waiting for a fix to merged into upstream mpifileutils. If this isn't merged soon, I will add skipForTicket and revisit this test case at a later time.

daltonbohning commented:
Current Status:
All test cases seem to be working, except for copy_subsets, pending a fix in upstream mpifileutils. I added skipForTicket until this is resolved.
I removed the Skip* tags, to ensure everything is in order.

daosbuild1 commented on src/tests/ftest/util/data_mover_utils.py:
```diff
@@ -0,0 +1,164 @@
+        self.timeout = timeout
+        self.hosts = hosts
+
+    def run(self, tmp, processes):
```
(pylint-arguments-differ) Arguments number differs from overridden 'run' method

daosbuild1 commented on src/tests/ftest/util/data_mover_test_base.py:
```diff
@@ -0,0 +1,398 @@
+        self.dm_cmd.get_params(self)
+        self.processes = self.params.get("np", '/run/datamover/processes/*')
+
+    # The valid paramter types for setting param locations.
```
```suggestion
    # The valid parameter types for setting param locations.
```

daosbuild1 commented on src/tests/ftest/util/data_mover_test_base.py:
```diff
@@ -0,0 +1,398 @@
+                UNS path in the dfuse mount.
+                Default is False.
+            dfuse_uns_pool (TestPool, optional): pool in the
+                dfuse mount for whcih to create a UNS path.
```
```suggestion
                dfuse mount for which to create a UNS path.
```

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3762/15/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/15/execution/node/58/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/15/execution/node/245/log

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/16/execution/node/58/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/16/execution/node/284/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/16/execution/node/280/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-3762/16/execution/node/287/log

daltonbohning commented:
Errors are being reported, but there is no indication of a failure.

daltonbohning commented:
Ubuntu build failed, which seems to have caused everything else to fail

daltonbohning commented:
Current Status:
All test cases seem to be working, except for copy_subsets, pending a fix in upstream mpifileutils. I added skipForTicket until this is resolved.
I removed the Skip* tags, to ensure everything is in order.

Ubuntu build failed due to missing rpm.
Added `PR-repos-el7` and `PR-repos-leap15` so Ubuntu doesn't try to build the missing rpm.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daltonbohning commented:
Currently, all tests are passing.

https://github.com/daos-stack/daos/pull/3332 was merged.
Nuking this PR, rebasing on master, and starting with a fresh PR.
